### ISSUE TITLE
Is there a use-case that allows for a client to upload their public key ?

### ISSUE THREAD

**Issue by nhi-vanye** [subspacecommunity/subspace#21]
Seems like having this server hold all the private keys is against the basics of Wireguard.

I'm interested in something that handles the peer files - but keeps the private keys on the physical device.

**Comment by agonbar** [comment]
It shouldn't be hard to do. Extending the client configuration to be able to change parameters like keys or IPs and save them. Not having the private key would make the download of a ready to use file impossible tough.

**Comment by nhi-vanye** [comment]
I agree that it impacts the one-stop shop, but it keeps the private keys where they were intended to be - on the device.

I don't see a reason to not support both use cases.

An alternative might be to destroy the private keys (securely) after they were used. That would mean that any subsequent update to the client's configuration would need a new private key generated, but that's probably good practice anyway (making it automatic/easy to rotate device keys - which is something I think tailscale does)

**Comment by gilesw** [comment]
I've been wondering about this. I don't want to have to issue all new keys if someone in systems leaves as they'll have had access to all the private keys. I've tested it out it doesn't blow up so I think I'll put a cronjob in place to run `sudo perl -pi -e 's/PrivateKey = .*/PrivateKey = YOU MUST MAKE A NEW PROFILE IF YOU LOST YOU KEY/' *.conf`
